include/deal.II/base/iterator_range.h

```diff
@@ -236,40 +174,148 @@ class IteratorRange
 };
 
 
+
+/**
+ * A class that implements the semantics of iterators over iterators as
+ * discussed in the design sections of the IteratorRange class.
+ */
+template <typename Iterator>
+class IteratorOverIterators
+{
+public:
+  /**
+   * Typedef the elements of the collection to give them a name that is more
+   * distinct.
+   */
+  using BaseIterator = Iterator;
+
+  /**
+   * Constructor. Initialize this iterator-over-iterator in such a way that
+   * it points to the given argument.
+   *
+   * @param iterator An iterator to which this object is supposed to point.
+   */
+  explicit IteratorOverIterators(const BaseIterator &iterator);
```

Comment:
Note that I marked this constructor as `explicit`. To my understanding, its highly unlikely that an instance of this class would ever be intentionally created outside of the context of an `IteratorRange`.

Reply:
Yes, this makes sense to mark it as `explicit`.